### ISSUE TITLE
Bug: Repository as npm package

### ISSUE THREAD

**Issue by Triloworld** [yashha/wpapi-extensions#9]
Hello, 
Error:
```
 ERROR  Failed to compile with 1 errors                                                                                                 friendly-errors 14:48:30

This dependency was not found:                                                                                                          friendly-errors 14:48:30
                                                                                                                                        friendly-errors 14:48:30
* wpapi-extensions in ./.nuxt/wp.js     
```

Scenario:
Found problem when testing : https://github.com/yashha/wp-nuxt/pull/109
If using branch or recursive mode (when is dependency of module with is in branch for testing) then error occur.

Where is problem:
This file don't exist in repository:
https://github.com/yashha/wpapi-extensions/blob/d4a441de5793c86db27d883c7cce6b9ebd998c71/package.json#L4

Solution temporary:
Go to wpapi-extensions directory in node_modules
Run: ```npm i``` 
Run: ```npm run build``` 
Now file exist and error go away.

Propose:
Add ```postinstall``` with ```npm install && npm run build``` to don't add file in repository but resolve bug in scenario of manually install

What you think @yashha ?

**Comment by yashha** [comment]
This could be fixed now with the latest release. 1.0.9

**Comment by Triloworld** [comment]
Thanks! Glad you have time to add small fixes

**Comment by Triloworld** [comment]
I get this when try local compile:

npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/babel-plugin-stage-2 - Not found
npm ERR! 404 
npm ERR! 404  'babel-plugin-stage-2@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
🚨  /wpapi-extensions/index.ts: Conflicting babel versions found in .babelrc. Make sure all of your plugins and presets depend on the same major version of babel.

**Comment by yashha** [comment]
I thought I updated to latest typescript?

**Comment by yashha** [comment]
How can I reproduce this?

**Comment by Triloworld** [comment]
You ave local one.
;( https://babeljs.io/blog/2018/07/27/removing-babels-stage-presets
They stop publish "stages" in babel. 

Parcel have new version and work:  
"parcel": "^2.0.0-alpha.3.2",
Check on fresh install without global " https://registry.npmjs.org/babel-plugin-stage-2 " as you have somewhere local

What you have there:
 https://registry.npmjs.org/babel-plugin-stage-2  ?

**Comment by Triloworld** [comment]
We waiting for response. It is critical for them as this dependency was rolled out.

We can wait for them for response if this is they side:
https://github.com/parcel-bundler/parcel/issues/4359
They will known faster why bug occur and how library can be changed :octocat:

**Comment by yashha** [comment]
I don't know, I can't reproduce it. But let's see what they will say.

**Comment by Triloworld** [comment]
Ok, babel is my fault. It was global scope where "stage-2" was in file. Now is in different, global scope and work like charm. 
Running "npm run build" don't resolve problem, try many different build scenario. It won't help if Parcel is in use and output used in webpack loader of nuxt.

Created https://github.com/yashha/wp-nuxt/pull/114 with fix this error once for all (detail there what change and why). Thanks for your help as I can, learning more what is going on. 

This one I will fix in pull request

**Comment by yashha** [comment]
I still don't know/when this problem occurs.
Can you give me a hint how to reproduce the error?
So I can check if the pr solves it?

What do you mean with
> If using branch or recursive mode (when is dependency of module with is in branch for testing) then error occur.

**Comment by Triloworld** [comment]
First step:
Start a nuxt project with WP as a backend, search for package to connect and decide to use this one. Run "npm install wpapi-extensions" next configuration and lastly run "npm run dev" in package.json as "nuxt".
It was "working" , in some time look up and found on compilation: "This dependency was not found:". 
So I think "maybe is with npm package something?". After all : It working so why error in first place?. I work on it in DEV mode, but SSR can be problematic (no documentation on this special mode and part of it isn't just visual plugin like carousel)

Next step:
Fix that one warning, easy? not in first contact.
Download package as a repository, put in directory in project dir (/local-files/wpapi-extensions), set package.json, and install manually by command "npm install". Next run "npm run build"
Running two other project fine and this stop on some cryptic message.
Now "not found" was real and webpack indicate it.
**Found that level up have ".babelrc" and It is catch by module.**
So this: "Conflicting babel versions found in .babelrc. " was resolved and file showed up 

> This part was my fault - sorry, but bad .babelrc was automatic fetch without my knowledge of Parcel built in mechanism

Main error to fix step:
We have same as npm package output. In meantime tested new version of Parcel with different scenario.
Parcel build output is not compatible with webpack and dual enviroment. One time webpack don't properly use, next browser have "not found" error. I try to get rid, in generate don't show up BUT on dev once more in "warn" version. 

Reproduce:
- copy github project to subfolder for some example nuxt instalation
- run "npm install" as normal
- get error

Experience:
- no instruction how I can help, 
- no automation in npm hooks
- error in first try as no easy indicate where is proper start to work on it

Solution was: https://github.com/yashha/wp-nuxt/pull/114 and only this place can designate place or this module will replace Parcel with webpack

Take that module as small project to fix bugs and work with it many times in future.
Propose was on start idea, next I get project in custom dir working, take test and now It is solution for narrow problem. 

With that can be edited README to instruct to run only "npm install" if you wan't to contribute.
Goal was to add (by myself) custom routes and fix bugs to have one, fully functional, production ready, WP REST API frontPage grabber.

**Comment by yashha** [comment]
I guess this is now fixed by #12, thank you very much

**Comment by Triloworld** [comment]
Ok. Thanks.
